tests/Vagrantfile

```diff
@@ -1,35 +1,38 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-$bootstrap = <<-EOF
+$centos9_bootstrap= <<-EOF
 #!/bin/bash
 set -e
-
-sudo cp /tmp/CentOS-Vault.repo /etc/yum.repos.d/
-sudo rm /etc/yum.repos.d/CentOS-Base.repo
-
-sudo yum install -y git python3
-python3 -m pip install --user --upgrade pip
-python3 -m pip install --user ansible==4.5.0
-sudo sed -i s/SELINUX=enforcing/SELINUX=disabled/ /etc/selinux/config
-echo Rebooting...
-sudo shutdown -r now
+sed -i s/SELINUX=enforcing/SELINUX=permissive/ /etc/selinux/config 
+setenforce permissive
+dnf install -y git python3.12
+python3.12 -m venv /opt/datalib_venv3.12
+/opt/datalib_venv3.12/bin/pip install --upgrade pip
+/opt/datalib_venv3.12/bin/pip install ansible==11.3.0 requests==2.32.3
```

Comment:
We used to just pin `ansible`, it worked for a while, then it stopped working. For now we can work around it by also pinning `requests`, but that won't work forever either. Someday another package will make a backwards-incompatible release and installs will stop working again until we pin it. The only reproducible way to use pip is to freeze the entire dependency list (`pip freeze > requirements.txt`, then build the environment with `pip install -r requirements.txt`.

We can defer this to another PR, but I would like to address it eventually.